### PR TITLE
BOCM-507 Issues for jest solved

### PR DESCRIPTION
**Pull request by MariaValeroGonzalezKC**
## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: apps/api/src/server/app/case-officer/appeal-formatter.js [hunk 1]
```diff
@@ -1,53 +1,50 @@
...
-	}
+	if (arrayOfStatusesContainsString(appealStatusesParallel,[appealStates.available_for_statements])) return 'available_for_statements';
+	if (arrayOfStatusesContainsString(appealStatusesParallel, [appealStates.available_for_final_comments])) return 'available_for_final_comments';
+	return '';
 }
 
 const appealFormatter = {
-	formatAppealForAllAppeals: function (appeal) {
+	/**
+	 * @param {{ appealStatus: object; id: any; reference: any; address: object; startedAt: any; }} appeal
```
Comment (mayya-bond-kc) [inline apps/api/src/server/app/case-officer/appeal-formatter.js:42]: Can we define a custom type for appealStatus as it seems we are repeating this definition a few times?

File: apps/api/src/server/app/case-officer/appeal-formatter.js [hunk 2]
```diff
@@ -56,8 +53,13 @@ const appealFormatter = {
 			QuestionnaireDueDate: appeal.startedAt ? formatDate(add2Weeks(appeal.startedAt)) : ''
 		};
 	},
-	formatAppealForParallelStates: function (appeal) {
+	/**
+	 * @param {{ appealStatus: object; id: any; reference: any; address: object; startedAt: any; }} appeal
+	 *  @returns {object}
```
Comment (mayya-bond-kc) [inline apps/api/src/server/app/case-officer/appeal-formatter.js:58]: I don't know if you want to define specifically what keys we can expect in the response?

File: apps/api/src/server/app/case-officer/case-officer.service.js [hunk 3]
```diff
@@ -1,23 +1,26 @@
...
-import { buildAppealCompundStatus } from '../utils/build-appeal-compound-status.js';
 import { breakUpCompoundStatus } from '../utils/break-up-compound-status.js';
+import { buildAppealCompundStatus } from '../utils/build-appeal-compound-status.js';
 
 /** @typedef {import('@pins/api').Schema.Appeal} Appeal */
 
-const reviewComplete = function (reviewReason) {
-	return Object.keys(reviewReason).every((index) => !reviewReason[index])? true : false;
+const reviewComplete = (reviewReason) => {
+	return !!Object.keys(reviewReason).every((index) => !reviewReason[index]);
```
Comment (mayya-bond-kc) [inline apps/api/src/server/app/case-officer/case-officer.service.js:10]: I think we can probably remove the `!!` here?

File: apps/api/src/server/app/inspector/appeal-formatter.js [hunk 4]
```diff
@@ -19,39 +19,22 @@ const provisionalAppealSiteVisitType = function (appeal) {
...
-		return addWeeksToDate(
-			getAppealStatusCreatedAt(appealStatus, 'available_for_final_comments'), 
-			weeksReceivingDocuments.finalComments
-		);	
-	} else {
-		throw new Error('Unknown status');
-	}
+const calculateExpectedSiteVisitBookingAvailableDate = (appealStatus) => {
+	if (arrayOfStatusesContainsString(appealStatus, 'available_for_statements')) return addWeeksToDate(getAppealStatusCreatedAt(appealStatus, 'available_for_statements'), weeksReceivingDocuments.statements + weeksReceivingDocuments.finalComments);
+	if (arrayOfStatusesContainsString(appealStatus, 'available_for_final_comments')) return addWeeksToDate(getAppealStatusCreatedAt(appealStatus, 'available_for_final_comments'), weeksReceivingDocuments.finalComments);
```
Comment (mayya-bond-kc) [inline apps/api/src/server/app/inspector/appeal-formatter.js:32]: These lines a too long

File: apps/api/src/server/app/state-machine/__tests__/full-planning-appeal.machine.test.js [hunk 5]
```diff
@@ -17,107 +16,109 @@ test.before('sets up mocking of actions', () => {
...
 	// 	if (context.inspectionType == 'accompanied' || context.inspectionType == 'access required') {
 	// 		sinon.assert.calledWithExactly(inspectorSendBookingStub, 1);
 	// 	} else {
 	// 		sinon.assert.notCalled(inspectorSendBookingStub);
 	// 	}
 	// }
-	// if (next_state.value == 'appeal_decided') {
+	// if (nextState.value == 'appeal_decided') {
 	// 	sinon.assert.calledWithExactly(notifyAppellantOfDecisionStub, 1, 'allowed');
 	// }
```
Comment (mayya-bond-kc) [inline apps/api/src/server/app/state-machine/__tests__/full-planning-appeal.machine.test.js:44]: do we need this block of code?

**Review by mayya-bond-kc — DISMISSED** [review]

**Comment by mayya-bond-kc** [comment]
I have noticed that you use a mix of arrow functions and regulardeclaration:
```
const funcName => (param) {...}
VS
function funcName(param) {...}
```
What is your thinking behind this?

File: apps/api/prisma/seed.js [hunk 6]
```diff
@@ -41,36 +50,36 @@ const appealTypes = {
...
 	completeValidationDecision = false,
 	lpaQuestionnaire = false,
-	startedAt = undefined,
 	incompleteReviewQuestionnaire = false,
 	completeReviewQuestionnaire = false,
 	connectToUser = false,
 	siteVisitBooked = false
-) {
+) => {
 	return {
```
[the remark this answers is not in the document]
Comment (MariaValeroGonzalezKC) [reply apps/api/prisma/seed.js]: this was corrected with a equal to null, and moving it back in place.. all working now

File: apps/api/src/server/app/case-officer/appeal-formatter.js (hunk 1 above)
Comment (MariaValeroGonzalezKC) [reply apps/api/src/server/app/case-officer/appeal-formatter.js:42]: this is now solved too

File: apps/api/src/server/app/case-officer/case-officer.service.js (hunk 3 above)
Comment (MariaValeroGonzalezKC) [reply apps/api/src/server/app/case-officer/case-officer.service.js:10]: correct, done!

File: apps/api/src/server/app/state-machine/__tests__/full-planning-appeal.machine.test.js (hunk 5 above)
Comment (MariaValeroGonzalezKC) [reply apps/api/src/server/app/state-machine/__tests__/full-planning-appeal.machine.test.js:44]: cleared!

**Pull request merged**